src/util/findRouteByComponent.js

```diff
@@ -1,22 +1,34 @@
-export default function findRouteByComponent(component, routes, components) {
+import isPlainObject from 'lodash.isplainobject';
+
+const isNamedComponents = isPlainObject;
+
+function searchNamedComponents(component, namedComponents, correspondingRoute, result) {
+  return Object.keys(namedComponents)
+    .some(name => {
```

Comment:
What do you think about using `find` here instead? That means that we can avoid having to pass along `result` and use the value directly?